url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/io/image/TargaImage.java

```diff
@@ -0,0 +1,345 @@
+//===========================================================================
+package vsdk.toolkit.io.image;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import java.awt.Dimension;
+import java.awt.image.DirectColorModel;
+
+import vsdk.toolkit.io.PersistenceElement;
+import vsdk.toolkit.media.RGBImage;
+import vsdk.toolkit.media.RGBAImage;
+
+/**
+The TargaImage class represents a targa image (the image files with .tga
+extension)
+*/
+
+public class TargaImage extends PersistenceElement
+{
+    private static final int NO_TRANSPARENCY = 255;
+    private static final int FULL_TRANSPARENCY = 0;
+    
+    private short idLength;
+    private short colorMapType;
+    private short imageType;
+    private int cMapStart;
+    private int cMapLength;
+    private short cMapDepth;
+    private int xOffset;
+    private int yOffset;
+    private int xSize;
+    private int ySize;
+    private short pixelDepth;
+    private short imageDescriptor;
+    private DirectColorModel cm;
+    public byte[] pixels;
+
+    /**
+    Constructs a TGAImage given the File image
+    @param srcFile The tga image file
+    */
+    public TargaImage(File srcFile) throws ImageNotRecognizedException {
+        try
+        {
+            open(srcFile);
+        }
+        catch(IOException ioe)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)